### PR TITLE
change flag name to better follow the other flags that start with disable

### PR DESCRIPTION
**Pull request by jessfraz**

### CODE REVIEW

**Comment by jfrazelle** [comment]
ping @tianon

File: docs/reference/commandline/daemon.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ weight=1
       --log-driver="json-file"               Default driver for container logs
       --log-opt=[]                           Log driver specific options
       --mtu=0                                Set the containers network MTU
-      --no-legacy-registry=false             Do not contact legacy registries
+      --disable-legacy-registry=false             Do not contact legacy registries
```
Comment (tianon) [inline docs/reference/commandline/daemon.md]: Alignment is a little off here. :smile:

**Comment by tianon** [comment]
As a total side note, hopefully v2 is never considered legacy or we're going to have a fun time with this flag name. :smile:

LGTM (minus minor formatting issue I commented on)

**Comment by jfrazelle** [comment]
also sorry I completely seemed to have missed this original PR and put my comments elsewhere when discussing it

**Comment by jfrazelle** [comment]
ok closing this was already decided on

**Comment by jfrazelle** [comment]
ping @docker/maintainers sorry I missed this original PR and tianon and I had discussed privately the opposition to the name but unfortunnately it never made it to the right place, so place your opinions here if you agree or disagree

**Comment by thaJeztah** [comment]
Absolutely! LGTM (I was actually thinking about this post-merge)

**Comment by thaJeztah** [comment]
Unless.. there are still people favoring https://github.com/docker/docker/pull/16385#issuecomment-141587844 :innocent:

**Comment by estesp** [comment]
I think `--disable-`.. aligns better with other flags; LGTM

**Comment by tianon** [comment]
Heh, "--enable" could be fun too -- I don't have strong feelings either way
personally.

**Comment by jfrazelle** [comment]
ya wrt --enable vs --disable im fine w either :)

**Comment by tiborvass** [comment]
If the goal is to keep v1 forever, then it should be `--enable-legacy-registry` that should default to true for now, and later on to false. If the goal is to get rid of v1 asap, then I agree with `--disable-legacy-registry` and make it a nop once we removed it. My understanding is we want the latter.

**Comment by jfrazelle** [comment]
i think we want v1 to burn in a fire no? :P

**Comment by cpuguy83** [comment]
Just as long as a Sith Lord doesn't find v1 all burned up, save v1 and put it in a suit and a special breathing apparatus.... that would be bad.

**Comment by icecrime** [comment]
LGTM

**Comment by thaJeztah** [comment]
docs LGTM

**Comment by jfrazelle** [comment]
ping @moxiegirl 0:)

File: contrib/completion/bash/docker [hunk 2]
```diff
@@ -526,7 +526,7 @@ _docker_daemon() {
 		--ip-masq=false
 		--iptables=false
 		--ipv6
-		--no-legacy-registry
+		--disable-legacy-registry
```
Comment (albers) [inline contrib/completion/bash/docker]: Please move the flag up so that it's alphabetically sorted.

Comment (jfrazelle) [inline contrib/completion/bash/docker]: Ah nice catch @albers! I will update!

Comment (jfrazelle) [inline contrib/completion/bash/docker]: updated there were a few spots :)

**Comment by thaJeztah** [comment]
ping @fredlf @jamtur01 @SvenDowideit @moxiegirl PTAL, think this is an easy pick :smile:

**Comment by jamtur01** [comment]
LGTM

**Comment by moxiegirl** [comment]
Thanks for the cover @jamtur01 Sorry gang, the cat has to go to the vet sometime and today was it.

**Comment by thaJeztah** [comment]
@moxiegirl no problem, just making sure we ping @jamtur01 once in a while, otherwise his LGTM's are getting rusty :stuck_out_tongue_winking_eye: 

(thanks James!)

**Comment by jamtur01** [comment]
You can ping me as often as you like. :)

**Pull request merged**